dpnp/backend/kernels/dpnp_krnl_sorting.cpp

```diff
@@ -83,6 +83,55 @@ struct _sort_less
     }
 };
 
+template <typename _DataType>
+class dpnp_partition_c_kernel;
+
+template <typename _DataType>
+void dpnp_partition_c(const void* sort_array1_in, void* result1, const size_t kth, const size_t* shape, const size_t ndim)
+{
+
+    cl::sycl::event event;
+
+    const _DataType* sort_arr = reinterpret_cast<const _DataType*>(sort_array1_in);
+    _DataType* result = reinterpret_cast<_DataType*>(result1);
+
+    size_t size_ = 1;
+    for (size_t i = 0; i < ndim - 1; ++i)
+    {
+        size_ *= shape[i];
+    }
+
+    cl::sycl::range<2> gws(size_, kth+1);
```

Comment:
In general case, the `shape[]` can contains zero as one of the dimension. It cause `size_` be zero. It will failed with GPU.
Please put "if()" to check GWS dimensions non zero in front of this line.